### PR TITLE
Refinements to Design and HTML

### PR DESCRIPTION
**Pull request by suneettipirneni**
- The right pane has been redesigned to make better use of it's real estate:
Screen Shot 2021-02-03 at 4 27 58 PM

- Corner radii have been all set consistently to 25px
- HTML is now fully adaptive and responds properly to many different window sizes
- Modals are no longer coded directly into the HTML file, they are dynamically added in when the page loads.
- Most of the CSS styling in the HTML files has been moved to the `styles.css file`

### CODE REVIEW

**Review by tylerhm — CHANGES_REQUESTED** [review]
This is awesome!

The new contact button is currently broken on my end though

**Comment by suneettipirneni** [comment]
> This is awesome!
> 
> The new contact button is currently broken on my end though

Weird, it's working for me, is your branch up to date?

**Comment by tylerhm** [comment]
> > This is awesome!
> > The new contact button is currently broken on my end though
> 
> Weird, it's working for me, is your branch up to date?

I'm pretty sure it is. Upon closer inspection it seems that the modals are not being pushed to the dom upon the page loading.

To try to reproduce the error, remove all of your contacts and reload the page.

**Comment by suneettipirneni** [comment]
> > > This is awesome!
> > > The new contact button is currently broken on my end though
> > 
> > 
> > Weird, it's working for me, is your branch up to date?
> 
> I'm pretty sure it is. Upon closer inspection it seems that the modals are not being pushed to the dom upon the page loading.
> 
> To try to reproduce the error, remove all of your contacts and reload the page.

You're right it doesn't work if you have no contacts, I'll look into that

File: js/contact.js [hunk 1]
```diff
@@ -77,26 +77,40 @@ function deselect(contactLink) {
...
-	document.getElementById("info-initials").innerHTML = contact.FirstName[0] + contact["LastName"][0];
-	document.getElementById("info-first-name").innerHTML = contact["FirstName"];
-	document.getElementById("info-last-name").innerHTML = contact["LastName"];
-	document.getElementById("info-phone-number").innerHTML = contact["PhoneNumber"];
-	document.getElementById("info-address").innerHTML = contact["Address"];
-	document.getElementById("info-zip").innerHTML = contact["ZIP"];
-	document.getElementById("info-city").innerHTML = contact["City"];
-	document.getElementById("info-state").innerHTML = contact["State"];
-
-	// Get editing ready
```
Comment (tylerhm) [inline js/contact.js:90]: Although I'm a little shady on what some of this does, it doesn't seem like the new loop manages the edit pane's placeholders.

Comment (suneettipirneni) [reply js/contact.js:90]: Good catch! I'll go ahead and add that back

**Comment by tylerhm** [comment]
@suneettipirneni  I've got a fix for the broken add button. In populateContacts, check to see if `fetchedContacts === undefined` before we run the .forEach on it

This was my bad a while back when first writing it.

**Comment by suneettipirneni** [comment]
> I've got a fix for the broken add button. In populateContacts, check to see if `fetchedContacts === undefined` before we run the .forEach on it

Yeah I saw that, however the root issue isn't the frontend, its the backend, it's returning a 400 for no contacts found when it should be returning a 200 because the query still was valid.

**Comment by tylerhm** [comment]
> > I've got a fix for the broken add button. In populateContacts, check to see if `fetchedContacts === undefined` before we run the .forEach on it
> 
> Yeah I saw that, however the root issue isn't the frontend, its the backend, it's returning a 400 for no contacts found when it should be returning a 200 because the query still was valid.

Oh gosh ok, gotta love api debugging ahaha

**Comment by suneettipirneni** [comment]
The add button should be fixed now

**Comment by tylerhm** [comment]
> The add button should be fixed now

Sweet!

File: js/contact.js (hunk 1 above)
Comment (tylerhm) [reply js/contact.js:90]: Once this is done go ahead and push it and I'll merge. This whole update really brings a layer of professionalism to the site, I'm loving it :D

**Review by tylerhm — APPROVED** [review]
Woo woo! Well done :) :+1:

**Pull request merged**